### PR TITLE
WIP: Edge signatures

### PR DESCRIPTION
**Pull request by Termina1**
Functions to sign dshackle responses when returned via gRPC.
Signs only if response has not already been signed.

- [x] Fix all tests
- [x] Add tests for signatures feature
- [x] Clean up
- [x] Make PR to Java API repository

### CODE REVIEW

**Comment by Termina1** [comment]
Waiting https://github.com/emeraldpay/emerald-grpc/pull/4 to be merged and propagated to Java API library

**Review by splix — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/emeraldpay/dshackle/config/SignatureConfig.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package io.emeraldpay.dshackle.config
+
+import java.security.PrivateKey
+
+class SignatureConfig {
+    /**
+     * Signature scheme that we should use
+     */
+    var signScheme: String = "SHA256withDSA"
```
[the remark this answers is not in the document]
Comment (Termina1) [reply src/main/kotlin/io/emeraldpay/dshackle/config/SignatureConfig.kt]: I made a enum with different sign scheme, but with another values

File: src/main/kotlin/io/emeraldpay/dshackle/config/SignatureConfig.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    var algorithm: String = "DSA"
```
[the remark this answers is not in the document]
Comment (Termina1) [reply src/main/kotlin/io/emeraldpay/dshackle/config/SignatureConfig.kt]: Made another enum for algoright, may be we would like to add some post quantum crypto as well =)

File: src/main/kotlin/io/emeraldpay/dshackle/config/SignatureConfig.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Termina1) [reply src/main/kotlin/io/emeraldpay/dshackle/config/SignatureConfig.kt]: Updated: removed signing schemes from config, now only sha256 is available

File: src/main/kotlin/io/emeraldpay/dshackle/rpc/CurrentResponseSigner.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import org.apache.commons.codec.binary.Hex
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Repository
+import java.security.Signature
+
+@Repository
+open class CurrentResponseSigner(
+    @Autowired private val config: SignatureConfig
+) : ResponseSigner {
+    override fun sign(message: ByteArray): String {
```
[the remark this answers is not in the document]
Comment (Termina1) [reply src/main/kotlin/io/emeraldpay/dshackle/rpc/CurrentResponseSigner.kt]: Now I directly pass nonce and message inside this method

**Pull request merged**